scipy/ndimage/filters.py

```diff
@@ -1187,6 +1187,9 @@ def rank_filter(input, rank, size=None, footprint=None, output=None,
     >>> ax2.imshow(result)
     >>> plt.show()
     """
+    if not isinstance(rank, int):
+        raise TypeError('Input rank needs to be integer')
+        return None
```

Comment:
In my opinion, there is no need `return` statement.
If raise condition is happened, the 'return' statement is unreachable.

Reply:
done